### DISCUSSION TITLE
Dynamic Partitions based on file's modified date

### DISCUSSION THREAD

**Discussion by ssillaots-boku** [dagster-io/dagster#19164] [Q&A]
Hey!
It seems to me that `has_dynamic_partition` checks only the file's existence.
I wonder if there will be development so that `has_dynamic_partition` also checks if/when file has been modified.

My ideal usecase:
* a file appears in AWS S3
* sensor picks it up and materializes it
* the same file gets overwritten in S3 meaning it will have a new `LastModified` value
* `has_dynamic_partitions` picks it up and it gets materialized again

**Comment by OwenKephart** [comment]
Hi @ssillaots-boku -- in this case, could you just make the dynamic partition key a union of the filename and the LastModified value? In dagster, the partition key is the unique identifier for a specific piece of data, so if you want two bits of data to be treated differently, they'll need different partition keys.

**Reply by ssillaots-boku** [reply]
Hi @OwenKephart. That's a great idea and solves the use case I wrote but actually my use case continues to redshift as well 😅 

So the problem arises when I want to backfill bunch of partitions. Let's say we have partitions `file_with_same_name-2022-02-02` and `file_with_same_name-2022-02-03`. First I got to make sure that `02` runs before `03`. Second it's just a waste of resources to run it two times when ideally only `03` would need to be rerun.

**Reply by clairelin135** [reply]
@ssillaots-boku 

In the pattern you're imagining below:
> a file appears in AWS S3
> sensor picks it up and materializes it
> the same file gets overwritten in S3 meaning it will have a new LastModified value
> has_dynamic_partitions picks it up and it gets materialized again

Could you instead modify the sensor to check for modifications? If the sensor recognizes that a file has been modified, it could kick off a new run request to materialize that same partition
